docs/tutorials/PyTorch.py

```diff
@@ -0,0 +1,322 @@
+#
+# This makes sense because we start at some x value and we want to get to a new x value that is closer to the minimum. In the above plot, the gradient is negative. If we were to add the gradient to our current x value that would bring us further away from the minimum to a more negative x value. This is because the gradient points in the direction of the steepest ascent while we are looking to go in the direction of the steepest descent. This is why there is a negative in the equation.
+#
+#
+# We will choose `step size = 0.1`:
+
+# +
+t = np.linspace(-5, 5, 100)
+plt.plot(t, t ** 2)
+
+
+plt.scatter(-4, 16)  # point
+
+plt.scatter(-3.2, 10.24)  # second point
+plt.arrow(-4, 16, 0.8, -5.76, width=0.07, color="r")
+plt.text(-4.5, 12, "step 1")
+
+plt.xlim(xmin=-5, xmax=5)
+plt.ylim(ymin=0, ymax=25)
+plt.show()
+# -
+
+# The gradient at our new point is still not close to zero, meaning we haven't reached the minimum. So we will take another step:
+
+# +
+t = np.linspace(-5, 5, 100)
+plt.plot(t, t ** 2)
+
+
+plt.scatter(-4, 16)  # point
+
+plt.scatter(-3.2, 10.24)  # second point
+plt.arrow(-4, 16, 0.8, -5.76, width=0.07, color="r")
+plt.text(-4.5, 12, "step 1")
+
+plt.scatter(-2.56, 6.5536)  # third point
+plt.arrow(-3.2, 10.24, 0.64, -3.6864, width=0.07, color="r")
+plt.text(-4, 7, "step 2")
+
+
+plt.xlim(xmin=-5, xmax=5)
+plt.ylim(ymin=0, ymax=25)
+plt.show()
+# -
+
+# We would continue this process of checking if the gradient is nearly zero, and taking a step in the direction of steepest descent until we reach the bottom of the valley. We'll say we've reached the bottom of the valley when the absolute value of the gradient is <0.1:
+
+# +
+t = np.linspace(-5, 5, 100)
+plt.plot(t, t ** 2)
+
+x_vals = np.array(
+    [
+        -4,
+        -3.2,
+        -2.56,
+        -2.048,
+        -1.6384,
+        -1.31072,
+        -1.048576,
+        -0.8388608,
+        -0.67108864,
+        -0.536870912,
+        -0.42949673,
+        -0.343597384,
+        -0.274877907,
+        -0.219902326,
+        -0.17592186,
+        -0.140737488,
+        -0.112589991,
+        -0.090071993,
+        -0.072057594,
+        -0.057646075,
+        -0.04611686,
+    ]
+)
+y_vals = np.array(
+    [
+        16,
+        10.24,
+        6.5536,
+        4.194304,
+        2.68435456,
+        1.717986918,
+        1.099511628,
+        0.703687442,
+        0.450359963,
+        0.288230376,
+        0.184467441,
+        0.118059162,
+        0.075557864,
+        0.048357033,
+        0.030948501,
+        0.019807041,
+        0.012676506,
+        0.008112964,
+        0.005192297,
+        0.00332307,
+        0.002126765,
+    ]
+)
+
+plt.scatter(x_vals, y_vals)
+
+plt.xlim(xmin=-5, xmax=5)
+plt.ylim(ymin=0, ymax=25)
+plt.show()
+# -
+
+# This works, but it takes a long time since we have several small steps. We could speed up the process by taking large steps. Starting at (-4,16) and increasing the step of size to 1.5 we find:
```

Comment:
I think you should de-emphasize (-4, 16) and just say something like "starting at the same initial location" since you are trying to highlight how changing the step size affects the process.